### PR TITLE
[shopsys] upgrade to Symfony 6.4

### PR DESCRIPTION
**Pull request by TomasLudvik**
#### Description, the reason for the PR
Symfony 5.4 lifecycle is ending so it is right time to upgrade to latest Symfony LTS.

#### Fixes issues
... 

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [ ] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

----
:globe_with_meridians: Live Preview:
  - https://tl-symfony-6-4.odin.shopsys.cloud
  - https://cz.tl-symfony-6-4.odin.shopsys.cloud

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -143,41 +143,41 @@
         "scheb/2fa-google-authenticator": "^6.12.0",
         "sentry/sentry-symfony": "^5.0.1",
         "shopsys/deployment": "^3.2.7",
-        "shopsys/jsformvalidator-bundle": "^1.7.0",
+        "shopsys/jsformvalidator-bundle": "2.0.x-dev",
```
Comment (vitek-rostislav) [inline composer.json]: please let's not forget to release the `2.0` version of the bundle and use it here

File: packages/administration/composer.json [hunk 2]
```diff
@@ -26,8 +26,8 @@
         "shopsys/framework": "16.0.x-dev",
         "shopsys/migrations": "16.0.x-dev",
         "shopsys/plugin-interface": "16.0.x-dev",
-        "symfony/config": "^5.4",
-        "symfony/dependency-injection": "^5.4",
+        "symfony/config": "^6.4",
+        "symfony/dependency-injection": "^6.4",
         "symfony/form": "^5.4",
```
Comment (vitek-rostislav) [inline packages/administration/composer.json]: why some of the packages are kept at `^5.4` version? I would like to upgrade them all :slightly_smiling_face:

**Comment by vitek-rostislav** [comment]
@coderabbitai review

File: project-base/app/tests/App/Test/WebTestCase.php [hunk 3]
```diff
@@ -248,4 +249,25 @@ protected function injectServices(): void
...
+        /** @var \Symfony\Component\HttpFoundation\RequestStack $requestStack */
+        $requestStack = static::$kernel->getContainer()->get(RequestStack::class);
+
+        $session = $sessionFactory->createSession();
+        $request->setSession($session);
+
+        $requestStack->push($request);
+
+        return $request;
+    }
```
Comment (vitek-rostislav) [inline project-base/app/tests/App/Test/WebTestCase.php:272]: could you please extract it into a helper, trait, common parent, or something like that so we avoid duplication with the code in `HttpSmokeTestCase`? :pray:

File: project-base/app/tests/App/Smoke/NewProductTest.php [hunk 4]
```diff
@@ -39,31 +37,24 @@ public function testCreateOrEditProduct(string $relativeUrl): void
...
-        /** @var \Symfony\Component\Security\Csrf\CsrfTokenManager $tokenManager */
-        $tokenManager = $client2->getContainer()->get('security.csrf.token_manager');
-        // if domain is on HTTPS, previously created token is prefixed with https-
-        $tokenId = ($isDomainSecured ? 'https-' : '') . ProductFormType::CSRF_TOKEN_ID;
-        $token = $tokenManager->getToken($tokenId);
-        $this->setFormCsrfToken($form, $token);
+        $em->beginTransaction();
 
-        $client2->submit($form);
+        $client->submit($form);
```
Comment (vitek-rostislav) [inline project-base/app/tests/App/Smoke/NewProductTest.php:52]: I am surprised it works even without the CSRF token :open_mouth:

Comment (vitek-rostislav) [inline project-base/app/tests/App/Smoke/NewProductTest.php:52]: I am surprised it works even without the CSRF token :open_mouth:

File: project-base/app/config/services.yaml [hunk 5]
```diff
@@ -528,6 +528,7 @@ services:
         arguments:
             - '@Redis'
             - { 'prefix': '%env(REDIS_PREFIX)%session:', ttl: 604800 }
+        lazy: true
```
Comment (vitek-rostislav) [inline project-base/app/config/services.yaml]: should not this whole service definition be in the framework package?

File: packages/framework/tests/Unit/Model/Security/LoginListenerTest.php [hunk 6]
```diff
@@ -10,61 +10,48 @@
...
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
-use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Event\LoginSuccessEvent;
 
 class LoginListenerTest extends TestCase
 {
     public function testOnSecurityInteractiveLoginUnique(): void
```
Comment (vitek-rostislav) [inline packages/framework/tests/Unit/Model/Security/LoginListenerTest.php]: `testOnSecurityInteractiveLoginUnique` and `testOnSecurityInteractiveLoginAdministrator` methods are now identical :slightly_smiling_face:

File: packages/framework/src/Form/Constraints/UniqueBillingAddress.php [hunk 7]
```diff
@@ -21,7 +21,7 @@ class UniqueBillingAddress extends Constraint
      * @var array<string, string>
      * @phpcsSuppress SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingNativeTypeHint
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Constraints/UniqueBillingAddress.php]: the suppress can be removed now, right? (it is on multiple places yet)

File: packages/framework/src/Form/Constraints/UniqueBillingAddress.php [hunk 8]
```diff
@@ -21,7 +21,7 @@ class UniqueBillingAddress extends Constraint
...
      */
-    protected static $errorNames = [
+    protected const array ERROR_NAMES = [
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Constraints/UniqueBillingAddress.php:23]: I see this change in +-2 classes only, what about the rest of the classes that extend `Symfony\Component\Validator\Constraint`?

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]

File: project-base/app/config/services.yaml (hunk 5 above)
Comment (TomasLudvik) [reply project-base/app/config/services.yaml]: Can be, it was not moved yet

File: project-base/app/tests/App/Test/WebTestCase.php (hunk 3 above)
Comment (TomasLudvik) [reply project-base/app/tests/App/Test/WebTestCase.php:272]: Those are a bit different use cases and without same parent I decided to keep it as it is.

Comment (vitek-rostislav) [reply project-base/app/tests/App/Test/WebTestCase.php:272]: ok :upside_down_face:

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**